chromium/background-scripts/background.js

```diff
@@ -114,6 +115,9 @@ if (chrome.windows) {
   chrome.windows.onFocusChanged.addListener(function() {
     updateState();
   });
+
+  // Grant access to HTTP site only during session, clear once window is closed
+  chrome.windows.onRemoved.addListener( () => httpOnceList.clear() );
```

Comment:
How would this behave with multiple windows? Shouldn't it apply for entire browsing session?

Reply:
I can try to manage by window ID and flag if the window is incognito I believe. However to fully manage sessions the extension would need the sessions permission, but I believe I can programmatically manage with the windows API alone.

Reply:
I think we should investigate split incognito mode. Regular and incognito parts of the extension can't communicate directly but they share storage.

Reply:
Agreed, will finish this piece up tomorrow.

Reply:
For now I have if all windows are closed regardless of type, since Incognito splitting proposed some issues I want to iron out.